### ISSUE TITLE
pip install - ValueError: dictionary update sequence element #0 has length 3; 2 is required

### ISSUE THREAD

**Issue by tseemann** [pysam-developers/pysam#349]
Installed via pip3 in virtualenv:

```
  Running setup.py (path:/tmp/pip-kb5lfk5k-build/setup.py) egg_info for package from file:///tmp/ariba--pysam-20160930-1
20692-1c7jwaf/pysam-0.9.1.4
    Running command python setup.py egg_info
    checking for gcc... /home/linuxbrew/.linuxbrew/bin/gcc-5 
checking whether the C compiler works... yes            
<snip>
config.status: creating config.mk                                                                                      
creating config.h
    # pysam: cython is available - using cythonize if necessary
    # pysam: htslib mode is shared
    # pysam: HTSLIB_CONFIGURE_OPTIONS=None
    # pysam: htslib configure options: --enable-libcurl
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-kb5lfk5k-build/setup.py", line 209, in <module>
             htslib_make_options = run_make_print_config()
      File "/tmp/pip-kb5lfk5k-build/setup.py", line 68, in run_make_print_config
      for line in stdout.splitlines()])
    ValueError: dictionary update sequence element #0 has length 3; 2 is required
Cleaning up...
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-kb5lfk5k-build/
```

**Comment by EricDeveaud** [comment]
this is due to the call to subprocess.check_output(["make", "print-config"])
line 69 in setup.py
when python setup.py is called by make, output of make print-config is the following
```
make[1]: Entering directory `/tmp/pysam-0.9.1.4/htslib'
LDFLAGS =
LIBHTS_OBJS = kfunc.o knetfile.o kstring.o bgzf.o faidx.o hfile.o hfile_net.o hts.o md5.o regidx.o sam.o synced_bcf_reader.o vcf_sweep.o tbx.o vcf.o vcfutils.o cram/cram_codecs.o cram/cram_decode.o cram/cram_encode.o cram/cram_external.o cram/cram_index.o cram/cram_io.o cram/cram_samtools.o cram/cram_stats.o cram/files.o cram/mFILE.o cram/open_trace_file.o cram/pooled_alloc.o cram/rANS_static.o cram/sam_header.o cram/string_alloc.o cram/thread_pool.o cram/vlen.o cram/zfio.o hfile_libcurl.o
LIBS = -lcurl -lcrypto
PLATFORM = Linux
make[1]: Leaving directory `/tmp/pysam-0.9.1.4/htslib'
```

make add "depth" of calls and make[1] breaks the split 

easy fix: edit setup.py and change 
```
subprocess.check_output(["make", "print-config"])
```

to 

```
subprocess.check_output(["make", "-s", "print-config"])

```

make will be silenced and will not output make[1]..... stuff.

**Comment by alexhenrie** [comment]
TLDR: If you call pysam's setup.py from a Makefile, you need to start make with `make -s`.

Another option would be to change pysam to parse and ignore make's extra output.

**Comment by AndreasHeger** [comment]
Thanks, added make -s as well.
